src/script/src/python/ffi_types/vector.rs

```diff
@@ -0,0 +1,524 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[cfg(test)]
+mod tests;
+use std::ops::Deref;
+use std::sync::Arc;
+
+use datatypes::arrow::array::{
+    Array, ArrayRef, BooleanArray, Float64Array, Int64Array, UInt64Array,
+};
+use datatypes::arrow::compute;
+use datatypes::arrow::compute::kernels::{arithmetic, comparison};
+use datatypes::arrow::datatypes::DataType as ArrowDataType;
+use datatypes::arrow::error::Result as ArrowResult;
+use datatypes::data_type::DataType;
+use datatypes::prelude::Value;
+use datatypes::value::{self, OrderedFloat};
+use datatypes::vectors::{Helper, NullVector, VectorRef};
+use pyo3::pyclass as pyo3class;
+use rustpython_vm::builtins::{PyBaseExceptionRef, PyBool, PyFloat, PyInt, PyNone};
+use rustpython_vm::sliceable::{SaturatedSlice, SequenceIndex, SequenceIndexOp};
+use rustpython_vm::types::PyComparisonOp;
+use rustpython_vm::{
+    pyclass as rspyclass, AsObject, PyObject, PyObjectRef, PyPayload, PyRef, PyResult,
+    VirtualMachine,
+};
+
+use crate::python::utils::is_instance;
+
+/// The Main FFI type `PyVector` that is used both in RustPython and PyO3
+#[pyo3class(name = "vector")]
+#[rspyclass(module = false, name = "vector")]
+#[derive(PyPayload, Debug, Clone)]
+pub struct PyVector {
+    pub(crate) vector: VectorRef,
+}
```

Comment:
Just wondering, do we need to add `#[repr(transparent)]` annotation to those python classes? This can allow us to treat a `PyVector` pointer as a `VectorRef` pointer.